lib/statsd_middleware.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+class StatsdMiddleware
+  def initialize(app)
+    @app = app
+  end
+
+  def call(env)
+    status, headers, response = @app.call(env)
+    params = env['action_dispatch.request.path_parameters']
+    key = "api.external.request#status=#{status},controller=#{params[:controller]},action=#{params[:action]}"
+    # rubocop:disable Lint/HandleExceptions
+    begin
+      StatsD.increment(key, 1)
+    rescue
+      # we want to ensure this doesn't break the response cyce
+    end
```

Comment:
Total nit, but you might be able to get away with just:

StatsD.increment(key, 1) rescue nil